lib/screenplay/prediction_suppression.ex

```diff
@@ -0,0 +1,96 @@
+defmodule Screenplay.PredictionSuppression do
+  @moduledoc """
+  Generates data structures for driving the prediction suppression UI
+  """
+  use GenServer
+
+  defguardp is_silver(route_id) when route_id in ["741", "742", "743", "746"]
+
+  @spec line_stops() :: [
+          %{
+            stop_id: String.t(),
+            line: String.t(),
+            direction_id: 0 | 1,
+            type: :start | :end | :mid
+          }
+        ]
+  def line_stops do
+    case :ets.lookup(:line_stops, :value) do
+      [{:value, data}] -> data
+      _ -> []
+    end
+  end
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, nil, name: __MODULE__)
+  end
+
+  @impl GenServer
+  def init(_) do
+    :ets.new(:line_stops, [:protected, :named_table, read_concurrency: true])
+    send(self(), :update)
+    {:ok, %{}}
+  end
+
+  @impl GenServer
+  def handle_info(:update, state) do
```

Comment:
A simple test case for this behavior would be nice to have, especially since it has no "real" users at the moment. We could stub a cut-down version of the API response (which still includes relevant edge cases), run the update, and assert that `line_stops` returns what the data should have been parsed into.

Reply:
I wanted to do this, but surprisingly the infrastructure for mocking HTTP responses isn't quite set up to make this straightforward. I may push that to a followup PR since it'll touch a bunch of things that aren't related. Also, the actual consumer of this data will be coming right after this.